showkase-component-impressions/src/main/java/com/vinaygaba/showkase_component_impressions/ComponentImpressions.kt

```diff
@@ -0,0 +1,145 @@
+package com.vinaygaba.showkase_component_impressions
+
+import android.annotation.SuppressLint
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.setValue
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.composed
+import androidx.compose.ui.layout.onGloballyPositioned
+import androidx.compose.ui.platform.LocalView
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.rememberUpdatedState
+import androidx.compose.ui.geometry.Rect
+import androidx.compose.ui.platform.LocalLifecycleOwner
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleEventObserver
+import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.flow.collect
+
+/**
+ * Passed in the callback of the [visibilityEvents] Modifier. Contains information about the
+ * visibility of a given composable function that uses the visibilityEvents Modifier.
+ */
+data class ShowkaseVisibilityEvent<T>(
+    val key: T,
+    val visibilityPercentage: Float,
+    val boundsInWindow: Rect
+)
+
+/**
+ * Use this modifier to get visibility events for a given Composable. It emits visibility events
+ * when the composable is added to the composition (visible), when its removed from the
+ * composition(invisible), when the activity is backgrounded(visible) and when the activity is
+ * foregrounded(visible). In addition,
+ *
+ * @param key Unique identifier for a given composable function that you use this modifier on
+ * @param onVisibilityChanged Callback that's called when the visibility of a composable function
+ * changes. This event has information about the key of the composable, the percentage of composable
```

Comment:
i think comments like `This event has information about the key of the composable, the percentage of composable
 * that's visible on the screen and the bounds of the composable.` can be omitted from here and instead should just be on that class